File: Bit/PlusOne/PlusOne.cc
class Solution {
	public:
		vector<int> plusOne(vector<int> &nums) {
			return plusN(nums, 1);
		}
		
		vector<int> plusN(vector<int> &nums, int n) {
			vector<int> result;
			int carry = n;
			for(int i = nums.size() - 1; i > 0; i--) {
				result.insert(result.begin(), (nums[i] + carry) % 10);
				carry = (nums[i] + carry) / 10;
			}
			if(carry)
				result.insert(result.begin(), carry);
			return result;
		}
};

File: Bit/PlusOne/PlusOne.java
class Solution {
	public int[] plusOne(int[] nums) {
		if(nums == null || nums.length == 0)
			return null;
		return plusN(nums, 1);
	}
		
	private int[] plusN(int[] nums, int n) {
		int[] result = new int[nums.length];
		int carry = n;
		for(int i = nums.length - 1; i > 0; i--) {
			result[i] = (nums[i] + carry) % 10);
			carry = (nums[i] + carry) / 10;
		}
		if(carry == 1) {
			int ret = new int[result.length + 1];
			ret[0] = 1;
			return ret;
		}
		return result;
	}
}